packages/commonwealth/client/scripts/views/pages/Editor/Editor.tsx

```diff
@@ -0,0 +1,262 @@
+  markdownShortcutPlugin,
+  MDXEditor,
+  MDXEditorMethods,
+  quotePlugin,
+  tablePlugin,
+  thematicBreakPlugin,
+  toolbarPlugin,
+} from 'commonwealth-mdxeditor';
+import React, { memo, useCallback, useRef, useState } from 'react';
+
+import './Editor.scss';
+
+import clsx from 'clsx';
+import 'commonwealth-mdxeditor/style.css';
+import { DesktopEditorFooter } from 'views/pages/Editor/DesktopEditorFooter';
+import { DragIndicator } from 'views/pages/Editor/DragIndicator';
+import { ToolbarForDesktop } from 'views/pages/Editor/ToolbarForDesktop';
+import { ToolbarForMobile } from 'views/pages/Editor/ToolbarForMobile';
+import { useEditorErrorHandler } from 'views/pages/Editor/useEditorErrorHandler';
+import { useImageUploadHandlerLocal } from 'views/pages/Editor/useImageUploadHandlerLocal';
+import { useImageUploadHandlerS3 } from 'views/pages/Editor/useImageUploadHandlerS3';
+import { codeBlockLanguages } from 'views/pages/Editor/utils/codeBlockLanguages';
+import { fileToText } from 'views/pages/Editor/utils/fileToText';
+import { iconComponentFor } from 'views/pages/Editor/utils/iconComponentFor';
+import supported from './markdown/supported.md?raw';
+
+export type ImageURL = string;
+
+export type EditorMode = 'desktop' | 'mobile';
+
+export type ImageHandler = 'S3' | 'local';
+
+type EditorProps = {
+  readonly mode?: EditorMode;
+  readonly placeholder?: string;
+  readonly imageHandler?: ImageHandler;
+};
+
+/**
+ * Handles supporting either of our image handlers.
+ */
+function useImageHandler(imageHandler: ImageHandler) {
+  const imageUploadHandlerDelegateLocal = useImageUploadHandlerLocal();
+  const imageUploadHandlerDelegateS3 = useImageUploadHandlerS3();
+
+  return useCallback(
+    async (file: File): Promise<ImageURL> => {
+      switch (imageHandler) {
+        case 'S3':
+          return await imageUploadHandlerDelegateS3(file);
+        case 'local':
+          return await imageUploadHandlerDelegateLocal(file);
+      }
+
+      throw new Error('Unknown image handler: ' + imageHandler);
+    },
+    [
+      imageHandler,
+      imageUploadHandlerDelegateLocal,
+      imageUploadHandlerDelegateS3,
+    ],
+  );
+}
+
+export const Editor = memo(function Editor(props: EditorProps) {
+  const errorHandler = useEditorErrorHandler();
+  const [dragging, setDragging] = useState(false);
+  const [uploading, setUploading] = useState(false);
+
+  const dragCounterRef = useRef(0);
+
+  const mode = props.mode ?? 'desktop';
+  const imageHandler: ImageHandler = props.imageHandler ?? 'S3';
+
+  const placeholder = props.placeholder ?? 'Share your thoughts...';
+
+  const mdxEditorRef = React.useRef<MDXEditorMethods>(null);
+
+  const imageUploadHandlerDelegate = useImageHandler(imageHandler);
+
+  const imageUploadHandler = useCallback(
+    async (file: File) => {
+      try {
+        setUploading(true);
+        return await imageUploadHandlerDelegate(file);
+      } finally {
+        setUploading(false);
+      }
```

Comment:
how do we handle the error case?

Reply:
I'm going to fix this in v1... I need to create a new handler function that just throws Error and verify it works.

The onError callback in MDXEditor is what should be called.